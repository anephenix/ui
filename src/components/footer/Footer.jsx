import React from 'react';
// Components

// Styling
import './Footer.scss';

const Footer = ({leftSection, rightSection }) => (
	<div id="footer">
		<div className='container'>
			<div className='withSidePadding'>
				{leftSection}
				{rightSection}
			</div>
		</div>
	</div>
);

export default Footer;
