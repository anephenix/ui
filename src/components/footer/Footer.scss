@import '../../../design-system/colours.scss';

#footer {
	margin-top: 100px;
	padding: 80px 10px 20px;
	display: flex;
	align-items: center;
	justify-content: center;
	font-size: 10pt;
	@media screen and (max-width: 767px) {
		padding: 80px 0px 20px;
	}

	.container {
		.withSidePadding {
			display: flex;
			flex-direction: row;
			align-items: center;
			justify-content: space-between;

			@media screen and (max-width: 767px) {
				flex-direction: column;
				justify-content: center;
				align-items: center;
				text-align: center;
			}

			a {
				color: $dark;

				@media screen and (prefers-color-scheme: dark) {
					color: $light;
				}

				text-decoration: none;
			}

			.made-in-location {
				display: flex;
				align-items: center;

				a {
					border-bottom: none;
				}

				@media screen and (max-width: 767px) {
					padding-top: 10px;
				}
			}
		}
	}
}
