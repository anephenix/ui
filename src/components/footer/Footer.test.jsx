import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Footer from './Footer';

describe('Footer', () => {
	test('renders correctly', () => {
		const { container } = render(<Footer />);
		expect(container).toMatchSnapshot();
	});

	test('displays the current year in the copright', () => {
		const { container } = render(<Footer />);
		const currentYear = new Date().getFullYear();
		const className = 'copyright';
		const element = container.querySelector(`.${className}`);
		expect(element).toBeTruthy();
		expect(element.textContent.match(new RegExp(currentYear))).toBeTruthy();
	});

	test('contains a link to Anephenix', () => {
		const { getByText } = render(<Footer />);
		const link = getByText('Anephenix').closest('a');
		expect(link).toHaveAttribute('href', 'https://anephenix.com');
		expect(link).toHaveAttribute('target', '_blank');
		expect(link).toHaveAttribute('rel', 'noopen noreferrer');
	});

	test('contains a link to the MIT License', () => {
		const { getByText } = render(<Footer />);
		const link = getByText('MIT License').closest('a');
		expect(link).toHaveAttribute('href', 'https://raw.githubusercontent.com/anephenix/ui/master/LICENSE');
		expect(link).toHaveAttribute('target', '_blank');
		expect(link).toHaveAttribute('rel', 'noopen noreferrer');
	});

	test('contains a link to the UK government site', () => {
		const { getByText } = render(<Footer />);
		const link = getByText('Made in the United Kingdom').closest('a');
		expect(link).toHaveAttribute('href', 'https://www.gov.uk');
		expect(link).toHaveAttribute('target', '_blank');
		expect(link).toHaveAttribute('rel', 'noopen noreferrer');
	});
});
