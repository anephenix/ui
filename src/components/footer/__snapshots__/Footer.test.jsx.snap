// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div>
  <div
    id="footer"
  >
    <div
      class="container"
    >
      <div
        class="withSidePadding"
      >
        <div
          class="copyright"
        >
          © 
          2025
           
          <a
            href="https://anephenix.com"
            rel="noopen noreferrer"
            target="_blank"
          >
            Anephenix
          </a>
          . UI is licensed under the
           
          <a
            href="https://raw.githubusercontent.com/anephenix/ui/master/LICENSE"
            rel="noopen noreferrer"
            target="_blank"
          >
            MIT License
          </a>
          .
        </div>
        <div
          class="made-in-location"
        >
          <svg
            height="12"
            viewBox="0 0 60 30"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <clippath
              id="s"
            >
              <path
                d="M0,0 v30 h60 v-30 z"
              />
            </clippath>
            <clippath
              id="t"
            >
              <path
                d="M30,15 h30 v15 z v15 h-30 z h-30 v-15 z v-15 h30 z"
              />
            </clippath>
            <g
              clip-path="url(#s)"
            >
              <path
                d="M0,0 v30 h60 v-30 z"
                fill="#012169"
              />
              <path
                d="M0,0 L60,30 M60,0 L0,30"
                stroke="#fff"
                stroke-width="6"
              />
              <path
                clip-path="url(#t)"
                d="M0,0 L60,30 M60,0 L0,30"
                stroke="#C8102E"
                stroke-width="4"
              />
              <path
                d="M30,0 v30 M0,15 h60"
                stroke="#fff"
                stroke-width="10"
              />
              <path
                d="M30,0 v30 M0,15 h60"
                stroke="#C8102E"
                stroke-width="6"
              />
            </g>
          </svg>
          <a
            href="https://www.gov.uk"
            rel="noopen noreferrer"
            target="_blank"
          >
              Made in the United Kingdom
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
